src/Paramore.Brighter.Outbox.Sqlite/SqliteOutboxSync.cs

```diff
@@ -174,6 +174,104 @@ public void Add(Message message, int outBoxTimeout = -1, IAmABoxTransactionConne
                 }
             }
         }
+        
+        /// <summary>
+        /// Awaitable add the specified message.
+        /// </summary>
+        /// <param name="messages">The message.</param>
+        /// <param name="outBoxTimeout">The time allowed for the write in milliseconds; on a -1 default</param>
+        /// <param name="transactionConnectionProvider">The Connection Provider to use for this call</param>
+        public void Add(IEnumerable<Message> messages, int outBoxTimeout = -1,
+            IAmABoxTransactionConnectionProvider transactionConnectionProvider = null)
+        {
+            var connectionProvider = _connectionProvider;
+            if (transactionConnectionProvider != null && transactionConnectionProvider is ISqliteTransactionConnectionProvider provider)
+                connectionProvider = provider;
+
+            var connection = connectionProvider.GetConnection();
+
+            if (connection.State != ConnectionState.Open) connection.Open();
+            var sql = GetBulkAddSql(messages.ToList());
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = sql.sql;
+                AddParamtersParamArrayToCollection(sql.parameters, command);
+
+                try
+                {
+                    if (connectionProvider.HasOpenTransaction)
+                        command.Transaction = connectionProvider.GetTransaction();
+                    command.ExecuteNonQuery();
+                }
+                catch (SqliteException sqlException)
+                {
+                    if (IsExceptionUnqiueOrDuplicateIssue(sqlException))
+                    {
+                        s_logger.LogWarning(
+                            "MsSqlOutbox: A duplicate Message was detected in the batch");
+                        return;
+                    }
+
+                    throw;
+                }
+                finally
+                {
+                    if(!connectionProvider.IsSharedConnection) connection.Dispose();
+                    else if (!connectionProvider.HasOpenTransaction) connection.Close();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Interface IAmABulkOutboxAsync
+        /// In order to provide reliability for messages sent over a <a href="http://parlab.eecs.berkeley.edu/wiki/_media/patterns/taskqueue.pdf">Task Queue</a> we
```

Comment:
We can probably drop the task queue from comments and replace with external bus. The task queue wording dates from early periods of Brighter when that was the major use case over microservice communication